File: 测试.py
#写一个python脚本在访问指定网址如'http://www.bilibili.com/video/'时，自动运行代码去修改html的内容，比如在<ul class="bpx-player-ctrl-playbackrate-menu">中添加一个新的<li class="bpx-player-ctrl-playbackrate-menu-item" data-value="0.25">0.25x</li>，使得在bilibili的html页面中可以选择0.25倍速播放。
import bpx
import player
from selenium import webdriver
from selenium.webdriver.edge.service import Service
from selenium.webdriver.edge.options import Options
import requests

if __name__ == '__main__':
    # 检测浏览器访问的网址是否是'http://www.bilibili.com/video/'
    # 如果是则自动运行代码去修改html的内容
    url = 'http://www.bilibili.com/video/'
    response = requests.get(url)
    if response.status_code == 200:
        html_content = response.text
        new_html_content = '<li class="bpx-player-ctrl-playbackrate-menu-item" data-value="0.25">0.25x</li>'
        html_content = html_content.replace('<ul class="bpx-player-ctrl-playbackrate-menu">', '<ul class="bpx-player-ctrl-playbackrate-menu">' + new_html_content)
        # 修改html内容
        response = requests.post(url, data=html_content)
        if response.status_code == 200:
            print('修改成功')
        else:
            print('修改失败')

File: 测试.js
// ==UserScript==
// @name         bilibil 倍速

// @version      0.3
// @description  按钮倍速
// @author       dfdy-yyc
// @match        https://www.bilibili.com/video/*
// @icon         https://www.google.com/s2/favicons?sz=64&domain=bilibili.com
// @grant        none
// ==/UserScript==

(function() {
    'use strict';
    console.log('123')
    window.onload=function() {
var intervalId = setInterval(function() {
    var parentElement = document.querySelector('.bpx-player-ctrl-playbackrate-menu');

    if (parentElement) {
        clearInterval(intervalId); // 如果找到元素，停止定时器
        console.log("找到元素:", parentElement);

        // 在这里编写需要执行的代码
         var parentElement1 = document.getElementsByClassName('bpx-player-ctrl-playbackrate-menu')[0]
        parentElement1.insertAdjacentHTML('afterbegin', '<li class="bpx-player-ctrl-playbackrate-menu-item" data-value="3.0">3.0x</li>');
    }

}, 100); // 每隔100毫秒执行一次搜索

    };})();




